pkg/client/request.go

```diff
@@ -478,6 +483,24 @@ func (r *Request) Do() Result {
 			continue
 		}
 
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			if retries == 10 {
+				// Give up after about 10 failed attempts.
+				return Result{respBody, created, err, r.codec}
+			}
+			retries++
+			if waitFor := resp.Header.Get("Retry-After"); waitFor != "" {
+				delay, err := strconv.Atoi(waitFor)
+				if err != nil {
+					// Give up on the retry if the delay is not an integer.
+					return Result{respBody, created, err, r.codec}
+				}
+				glog.Infof("Got a Retry-After %s response for attempt %d to %v", waitFor, retries, r.finalURL())
```

Comment:
This should be a verbose log. Connection and rate-limit problems are commonplace. Let's not spam the logs.